src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleObject.cs

```diff
@@ -96,7 +97,20 @@ private static bool WriteEndObject(ref WriteStack state)
 
             if (jsonPropertyInfo.ClassType == ClassType.Value)
             {
-                jsonPropertyInfo.Write(ref state, writer);
+                bool includeProperty = true;
```

Comment:
We need to consider properties of other [class types](https://github.com/dotnet/corefx/blob/cb5be3e9c76c337cb1752b206a50ffc71947aba1/src/System.Text.Json/src/System/Text/Json/Serialization/ClassType.cs) as well.